js_modules/dagster-ui/packages/ui-core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -537,7 +537,9 @@ async function stateForLaunchingAssets(
         assetSelection: assets.map((a) => ({assetKey: a.assetKey, opNames: a.opNames})),
         assetChecksAvailable: assets.flatMap((a) =>
           a.assetChecksOrError.__typename === 'AssetChecks'
-            ? a.assetChecksOrError.checks.map((check) => ({...check, assetKey: a.assetKey}))
+            ? a.assetChecksOrError.checks
+                .filter((check) => check.jobNames.includes(jobName))
```

Comment:
I'm not familiar with the logic here but should this be an `includes` as opposed to an exact match? Does it rely on the assumption that a a user won't use the same name for a check and a job?